NitroxClient/GameLogic/Spawning/WorldEntities/PlaceholderGroupWorldEntitySpawner.cs

```diff
@@ -7,75 +7,93 @@
+            {
+                IPrefabRequest prefabCoroutine = PrefabDatabase.GetPrefabAsync(prefabPlaceholder.prefabClassId);
+                yield return prefabCoroutine;
+                prefabCoroutine.TryGetPrefab(out GameObject prefab);
+                GameObject gameObject = Utils.SpawnZeroedAt(prefab, prefabPlaceholder.transform, true);
 
-                if (child is PrefabPlaceholderEntity placeholder)
-                {
-                    PrefabPlaceholder prefabPlaceholder = prefabPlaceholderGroup.prefabPlaceholders[index];
+                NitroxEntity.SetNewId(gameObject, placeholder.Id);
 
-                    IPrefabRequest prefabCoroutine = PrefabDatabase.GetPrefabAsync(prefabPlaceholder.prefabClassId);
-                    yield return prefabCoroutine;
-                    prefabCoroutine.TryGetPrefab(out GameObject prefab);
-                    GameObject gameObject = Utils.SpawnZeroedAt(prefab, prefabPlaceholder.transform, true);
+                Optional<EntityMetadataProcessor> metadataProcessor = EntityMetadataProcessor.FromMetaData(entity.Metadata);
 
-                    NitroxEntity.SetNewId(gameObject, placeholder.Id);
+                if (metadataProcessor.HasValue)
+                {
+                    metadataProcessor.Value.ProcessMetadata(gameObject, placeholder.Metadata);
+                }
+            }
+            else if (child is WorldEntity slotEntity)
+            {
+                IWorldEntitySpawner spawner = spawnerResolver.ResolveEntitySpawner(slotEntity);
+                Optional<GameObject> slotEntityParent = Optional.Of(prefabPlaceholder.gameObject);
 
-                    Optional<EntityMetadataProcessor> metadataProcessor = EntityMetadataProcessor.FromMetaData(entity.Metadata);
+                TaskResult<Optional<GameObject>> slotEntityTaskResult = new();
+                yield return spawner.SpawnAsync(slotEntity, slotEntityParent, cellRoot, slotEntityTaskResult);
 
-                    if (metadataProcessor.HasValue)
-                    {
-                        metadataProcessor.Value.ProcessMetadata(gameObject, placeholder.Metadata);
-                    }
-                }
-                else
+                // Resetting local position
+                if (slotEntityTaskResult.Get().HasValue)
                 {
-                    Log.Debug($"Unhandled child type {child}");
-                    continue;
+                    Transform slotTransform = slotEntityTaskResult.Get().Value.transform;                
+                    slotTransform.localPosition = Vector3.zero;
+                    slotTransform.localRotation = Quaternion.identity;
                 }
             }
+            else
+            {
+                Log.Debug($"Unhandled child type {child}");
+            }
         }
+    }
 
-        public bool SpawnsOwnChildren()
-        {
-            return true;
-        }
-}
-
+    public bool SpawnsOwnChildren()
+    {
+        return true;
```

Comment:
One idea I had is to introduce a new spawner for the placeholder children, called PrefabPlaceholderEntitySpawner.  That way we do not need to spawn our own children and if it is a world entity then it will automatically be routed to the correct WorldEntity spawner.  Then the logic here is very simple: spawn the parent, tag it with nitrox id, and allow children to be spawned upstream.

Reply:
Problem is that we need the group to get the correct parent and position. We can not completely handle even the world entities independent from the group.

Reply:
Is this because we need to do the slot transformation after the item is spawned (lines 83-85 above)?  Is it possible that we can do this as a pre-processing step when creating the entity and not at spawn time?

Reply:
It's not 83-85, that is just a fix for a strange bug.
Ideally prefabs are spawned parented to the corresponding PrefabPlaceholder. Problem is that the server doesn't know about these placeholders nor the hierarchy.